### PR TITLE
cleanup: remove dead SDPA patches

### LINKED ISSUE #3331: [ Cleanup ] Remove dead sdpa patches
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Transformers use https://github.com/huggingface/transformers/blob/main/src/transformers/integrations/sdpa_attention.py#L40

and does not use the 4D mask convert which we patched anymore

- src/axolotl/monkeypatch/llama_patch_multipack.py
- and others

### ✔️ Solution

Delete

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by OnePunchMonk**
Transformers 5.x routes attention through sdpa_attention.py and no longer calls the _prepare_4d_causal_attention_mask* or _expand_mask functions that these patches targeted. This makes the following patches dead code:

- llama_patch_multipack.py (patched _prepare_4d_causal_attention_mask*)
- llama_expand_mask.py (patched _expand_mask, never called)
- Related utility functions in monkeypatch/utils.py

Closes axolotl-ai-cloud/axolotl#3331

# Description

## AI Usage Disclaimer
Claude

## Summary by CodeRabbit

* **Refactor**
  * Removed Llama model-specific attention mask patches and related utility functions.
  * Removed expand_mask test suite and configuration entries.

### CODE REVIEW

**Comment by OnePunchMonk** [comment]
@coderabbitai review

**Review by NanoCode012 — APPROVED** [review]

**Comment by NanoCode012** [comment]
Thank you. Less code is great!
